### DISCUSSION TITLE
Strange Sfdump-script with 3512 lines in my VirtualHost webpage

### DISCUSSION THREAD

**Discussion by comprog-ux** [symfony/symfony#46487] [Q&A]
In this: http://shooter.mywire.org you can see three parts of the script. I don't know what's causing this.
If I do a Show Page Source the first few lines look like this:
```
<script>
Sfdump = window.Sfdump || (function(doc) {
    var refStyle = doc.createElement('style'),
        rxEsc = /([.*+?^${}()|\[\]\/\\])/g,
        idRx = /\bsf-dump-\d+-ref[012]\w+\b/,
        keyHint = 0 <= navigator.platform.toUpperCase().indexOf('MAC') ? 'Cmd' : 'Ctrl',
        addEventListener = function(e, n, cb) {
            e.addEventListener(n, cb, false);
        };
    refStyle.innerHTML = 'pre.sf-dump .sf-dump-compact, .sf-dump-str-collapse .sf-dump-str-collapse, .sf-dump-str-expand .sf-dump-str-expand { display: none; }';
    (doc.docu...................
```
It's a Laravel9 system running on Ubuntu 18.04 system with PHP 8.1.6 (cli). The shooter.mywire.org is a ddns from dynu.com and created by IspConfig.

The system have Symfony/Console installed but not Symfony.

What could be wrong?

**Comment by alessandro-podo** [comment]
it looks like Code from symfony/web-profiler-bundle
Do you use this Package?

**Reply by comprog-ux** [reply]
Thanks, but there were something wrong with the code. I cloned down fresh code from github and after that it works fine. Earlier I had taken the code from my working folder in my mac. That wasn't good apparently. This is the working site: http://shoot.ddnsfree.com

**Comment by comprog-ux** [comment]
I have another problem though if you can look at that: https://github.community/t/want-to-restrict-members-in-an-invoice-laravel9/253367?fbclid=IwAR2oiH0hkDKV2azq9_sMwOhNXbxKH4lv_g2TmgX9NDcOq4trzIr2-KvZvrY

**Comment by yannhodiesne** [comment]
For the record for those who end up here after a Google search, whenever a response begins with `<script> Sfdump = window.Sfdump || ...` it means you forgot to comment or delete a `dump(...)` statement in your codebase.

For whatever reason Symfony does not do anything with this `dump` call in a development environment, but if `APP_ENV` is set to `prod` it returns this gargantuan mess of a script instead of your JSON response.

Source: https://github.com/symfony/symfony/discussions/44993

**Reply by stof** [reply]
Well, in the dev environment, you generally have DebugBundle that configures `dump` to store the dump in the web profiler data instead of dumping it directly in the response.
In prod, both DebugBundle and the web profiler are disabled, so the dump behavior is the default one, dumping to the output (just like what `var_dump` would do all the time)

**Reply by yannhodiesne** [reply]
Oh thanks for the explanation, it makes a ton of sense from the framework's perspective!

I'm still not a huge fan of having a potentially "response corrupting" default behavior sneaking up on you by showing itself only in production, but I'll blame this on personal preferences and my lack of experience with Symfony

**Reply by stof** [reply]
Well, if you put a `var_dump` in your production code, you also get a response corrupting behavior.

What I'm using myself is that I have a php_code_sniffer rule that forbids the `dump()` function, so the CI will not let me merge anything with a `dump()` so it won't reach production.

And if you don't do that, you could configure your prod environment with a blackhole dumper that would ignore any dumped value
